src/bin/pg_autoctl/keeper.c

```diff
@@ -483,6 +484,25 @@ keeper_update_pg_state(Keeper *keeper)
 	{
 		/* Postgres is not running. */
 		postgres->pgIsRunning = false;
+
+		/* keep the current values we have for the Postgres characteristics */
+		if (pgSetup->control.pg_control_version != 0)
```

Comment:
Shouldn't this be `keeperState->pg_control_version`?

Reply:
I believe you're right. Cache invalidation still is in the top-3 of difficult things, right?